windows/vNext/ReactNativeVideoCPP/ReactVideoView.cpp

```diff
@@ -0,0 +1,255 @@
+                {
+                  eventDataWriter.WriteObjectBegin();
+                  WriteProperty(eventDataWriter, L"width", width);
+                  WriteProperty(eventDataWriter, L"height", height);
+                  WriteProperty(eventDataWriter, L"orientation", orientation);
+                  WriteProperty(eventDataWriter, L"orientation", orientation);
+                  eventDataWriter.WriteObjectEnd();
+                }
+
+                WriteProperty(eventDataWriter, L"canPlayFastForward", false);
+                WriteProperty(eventDataWriter, L"canPlaySlowForward", false);
+                WriteProperty(eventDataWriter, L"canPlaySlow", false);
+                WriteProperty(eventDataWriter, L"canStepBackward", false);
+                WriteProperty(eventDataWriter, L"canStepForward", false);
+              }
+              eventDataWriter.WriteObjectEnd();
+            });
+      }
+    }
+  });
+}
+
+void ReactVideoView::OnMediaFailed(IInspectable const &, IInspectable const &) {}
+
+void ReactVideoView::OnMediaEnded(IInspectable const &, IInspectable const &) {
+  runOnQueue([weak_this{get_weak()}]() {
+    if (auto strong_this{weak_this.get()}) {
+      strong_this->m_reactContext.DispatchEvent(*strong_this, L"topEnd", nullptr);
+    }
+  });
+}
+
+void ReactVideoView::OnBufferingStarted(IInspectable const &, IInspectable const &) {}
+
+void ReactVideoView::OnBufferingEnded(IInspectable const &, IInspectable const &) {}
+
+void ReactVideoView::OnSeekCompleted(IInspectable const &, IInspectable const &) {
+  runOnQueue([weak_this{get_weak()}]() {
+    if (auto strong_this{weak_this.get()}) {
+      strong_this->m_reactContext.DispatchEvent(*strong_this, L"topSeek", nullptr);
+    }
+  });
+}
+
+void ReactVideoView::Set_progressUpdateInterval(int64_t interval) {
+  m_timer.Interval(std::chrono::milliseconds{interval});
+}
+
+void ReactVideoView::Set_IsLoopingEnabled(bool value) {
+  m_isLoopingEnabled = value;
+  if (m_player != nullptr) {
+    m_player.IsLoopingEnabled(m_isLoopingEnabled);
+  }
+}
+
+void ReactVideoView::Set_UriString(hstring const &value) {
+  m_uriString = value;
+  if (m_player != nullptr) {
+    auto uri = Uri(m_uriString);
+    m_player.Source(MediaSource::CreateFromUri(uri));
+  }
+}
+
+void ReactVideoView::Set_Paused(bool value) {
+  m_isPaused = value;
+  if (m_player != nullptr) {
+    if (m_isPaused) {
+      if (IsPlaying(m_player.PlaybackSession().PlaybackState())) {
+        m_player.Pause();
+      }
+    } else {
+      if (!IsPlaying(m_player.PlaybackSession().PlaybackState())) {
+        m_player.Play();
+      }
+    }
+  }
+}
+
+void ReactVideoView::Set_Muted(bool isMuted) {
+  m_isMuted = isMuted;
+  if (m_player != nullptr) {
+    m_player.IsMuted(m_isMuted);
+  }
+}
+
+void ReactVideoView::Set_Controls(bool useControls) {
+  m_useControls = useControls;
+  AreTransportControlsEnabled(m_useControls);
+}
+
+void ReactVideoView::Set_FullScreen(bool fullScreen) {

```

Comment:
When you set the video to fullscreen can you still receive keydown events from the <Video /> in js?
For my private (and less complete :)) video viewmanager I found that I had to handle that specifically as the MediaPlayerElement gets moved into the FullScreenMediaRoot tree, and is no longer a parent of the react root view.
Maybe something else takes care of this or I was missing something, but figured it would be good to ask :D

Reply:
We need to fix this in RNW code by specializing for MediaPlayerElement. This also impacts touch events too, we need to do similar think as we have done for flyout and popup. I will create an issue in RNW and fix it there.

Reply:
Created RNW issue https://github.com/microsoft/react-native-windows/issues/4157.